common/io/src/marshal.rs

```diff
@@ -117,3 +117,13 @@ impl Marshal for bool {
         scratch[0] = *self as u8;
     }
 }
+
+impl Marshal for char {
+    fn marshal(&self, scratch: &mut [u8]) {
+        let bits = *self as u32;
+        scratch[0] = bits as u8;
+        scratch[1] = (bits >> 8) as u8;
+        scratch[2] = (bits >> 16) as u8;
+        scratch[3] = (bits >> 24) as u8;
+    }
```

Comment:
Does this marshaling preserve the order before and after marshaling?

Reply:
Yup, `char` is just `u32` in Rust

Reply:
I mean...does this hold: `char_a > char_b ↔ marshal(char_a) > marshal(char_b)`?
Because we need to list tenants in alphabetic order.

```rust
fn marshal(c: char) -> Vec<u8> {
    let mut scratch = vec![0, 0, 0, 0];

    let bits = c as u32;
    scratch[0] = bits as u8;
    scratch[1] = (bits >> 8) as u8;
    scratch[2] = (bits >> 16) as u8;
    scratch[3] = (bits >> 24) as u8;

    scratch
}

fn main() {
    let a: char = '我';
    let b: char = '发';

    let aa = marshal(a);
    let bb = marshal(b);

    println!("{} {} {:?} {:?} ", a, b, aa, bb);

    assert_eq!(a > b, aa > bb);
}

```

```
我 发 [17, 98, 0, 0] [209, 83, 0, 0] 
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`', src/main.rs:43:5
```